client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -54,7 +55,8 @@ export default function StepStoreAddress( props: WooCommerceInstallProps ): Reac
 	} );
 
 	const { get, save, update } = useSiteSettings( siteId );
-	const domain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
+
+	const backUrl = signupDependencies.back_to;
```

Comment:
```suggestion
	const domain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
	const backUrl = signupDependencies?.back_to || `/woocommerce-installation/${ domain }`;
```

Reply:
This backUrl should be limited to relative paths.

e.g. This should not work and just ignore it if it's invalid:

```
http://calypso.localhost:3000/start/woocommerce-install/store-address?site=test803879000.wordpress.com&back_to=https://example.com
``` 

Something like this should be enough:
https://github.com/Automattic/wp-calypso/blob/9cf831eeb055e5a65784011fe1337be388658281/client/my-sites/checkout/composite-checkout/lib/leave-checkout.ts#L68

This also blocks less than ideal urls like 
```
http://calypso.localhost:3000/start/woocommerce-install/store-address?site=test803879000.wordpress.com&back_to=javascript:alert(%27sup%27);
```

Reply:
> This backUrl should be limited to relative paths.

Yep looks like we lost that validation in https://github.com/Automattic/wp-calypso/pull/60892/commits/4b2ce06ca91fc575b74bf9e060ecbe2414720aee, let's add that back.

Reply:
Sorry got a little confused on what that code was doing and removed it my bad there. I think maybe this a good place adding a comment to put some light on what the regexp does.

Reply:
That's a good idea 👍

Reply:
Yea, that code really needs to be made a function, there are a few similar/exact copies of it around.

Reply:
Ahhh it would be nice to replace that. 

I added the validation and fallback here: https://github.com/Automattic/wp-calypso/pull/60892/commits/fc3f13356ceaf2b573f3bfeaf83b65e9bb8b70e4